docs/graphql/manual/actions/codegen/examples/python-flask/index.rst

```diff
@@ -0,0 +1,387 @@
+
+.. code-block:: python
+
+  from argon2 import PasswordHasher
+  Password = PasswordHasher()
+
+  @app.route("/signup", methods=["POST"])
+  def signup_handler():
+      args = AuthArgs.from_request(request.get_json())
+      hashed_password = Password.hash(args.password)
+
+Creating the GraphQL request client
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Next, since we have the user's email and hashed password, we need to send a request to Hasura to save them in the database. For that, we'll need a request client implementation:
+
+.. code-block:: python
+
+  from requests import request
+
+  @dataclass
+  class Client:
+      url: str
+      headers: dict
+
+      def run_query(self, query: str, variables: dict, extract=False):
+          request = requests.post(
+              self.url,
+              headers=self.headers,
+              json={"query": query, "variables": variables},
+          )
+          assert request.ok, f"Failed with code {request.status_code}"
+          return request.json()
+
+      create_user = lambda self, email, password: self.run_query(
+          """
+              mutation CreateUser($email: String!, $password: String!) {
+                  insert_user_one(object: {email: $email, password: $password}) {
+                      id
+                      email
+                      password
+                  }
+              }
+          """,
+          {"email": email, "password": password},
+      )
+
+Here we create a utility class for handling our Hasura operations. It takes an URL and headers object as initialization options, and exposes a method ``.run_query()`` for performing GraphQL requests. We create the query function for saving our user in the ``Signup`` action as a class method as well.
+
+We can instantiate the ``Client`` like this:
+
+.. code-block:: python
+
+  HASURA_URL = "http://graphql-engine:8080/v1/graphql"
+  HASURA_HEADERS = {"X-Hasura-Admin-Secret": "your-secret"}
+
+  client = Client(url=HASURA_URL, headers=HASURA_HEADERS)
+
+Now, in our ``Signup`` action handler, we need to call ``client.create_user()`` with the input email and the hashed password value to save them, then return the result:
+
+.. code-block:: python
+
+  @app.route("/signup", methods=["POST"])
+  def signup_handler():
+      args = AuthArgs.from_request(request.get_json())
+      hashed_password = Password.hash(args.password)
+      user_response = client.create_user(args.email, hashed_password)
+      if user_response.get("errors"):
+          return {"message": user_response["errors"][0]["message"]}, 400
+      else:
+          user = user_response["data"]["insert_user_one"]
+          return CreateUserOutput(**user).to_json()
+
+To test this out, send an HTTP request to your Flask API at ``/signup`` with an email and password. You should get a successful response like this:
+
+.. thumbnail:: ../../../../../../img/graphql/manual/actions/python-flask-signup-request.png
```

Comment:
@GavinRay97 can we have this as a code snippet? It's nice if users can copy-paste the code to try this out.